src/payload_storage.rs

```diff
@@ -105,14 +111,14 @@ impl PayloadStorage {
         if self.pages.is_empty() {
             return None;
         }
-        let needed_size =
-            SlottedPageMmap::MIN_VALUE_SIZE_BYTES.max(payload_size) + SlotHeader::size_in_bytes();
```

Comment:
fixed encapsulation issue, the storage layer should not know about the slot sizes